### PR TITLE
Add custom base layer

### LINKED ISSUE #74: Add your own base layer
Now that @lyzidiamond added the feature to switch base layers, it would be awesome to implement a custom layer. This would essentially allow the user to input their own Mapbox tile ID and key?

@lyzidiamond do you know if this would require a key or would the ID be sufficient?

### PR DESCRIPTION
**Pull request by hamhands**
### What changed
* added custom base map button
* takes map id and layer name

### Issues
* Closes #74 

### What's needed 
* way to select the layer once it's added

### Screencap
custom_base

### CODE REVIEW

**Comment by powersa** [comment]
Make this the default!

**Comment by alukach** [comment]
:suspect: :rage1: :rage2: :rage3: :rage4: :gobeserk:

Love it, great work.

File: src/js/ops.file.dropchop.js [hunk 1]
```diff
@@ -157,6 +157,47 @@ var dropchop = (function(dc) {
...
+          var newLayer = dc.map.baseLayers.addBaseLayer(L.mapbox.tileLayer(dc.ops.file['load-custom-base']._temp),
+            dc.ops.file['load-custom-base']._temp_1);
+          L.mapbox.tileLayer(newLayer).addTo(dc.map.m);
+        } else {
+          dc.notify('error', xhr.status + ': could not retrieve base map. Please check your Map ID');
+        }
+      }
+    },
+
+    'break1': { type: 'break' },
```
Comment (alukach) [inline src/js/ops.file.dropchop.js]: Not sure if this is needed anymore, as we manually construct the menu now.

**Comment by jczaplew** [comment]
It would be nice to be able to add custom base maps from sources other than Mapbox as well. Perhaps we could check for the presence of `http` in the provided URL/map ID and display accordingly? 

Also, since these things are URL-based we could definitely store this state in the URI like we do for gists and layers from URLs

**Comment by mapsam** [comment]
Also, is codeship broken here @alukach? I see all of the open PRs with nothing but nothin on the codeship side.

File: src/js/ops.file.dropchop.js (hunk 1 above)
Comment (mapsam) [inline src/js/ops.file.dropchop.js]: Are you talking specifically about the `break` piece in there?

Comment (alukach) [inline src/js/ops.file.dropchop.js]: Yup, just that part.

Comment (mapsam) [inline src/js/ops.file.dropchop.js]: Doh, sorry didn't realize this was a line comment.

**Comment by alukach** [comment]
@mapsam not sure what's up with codeship here... seems like it didn't even try to build. If I had to guess, I'd say that it was because this is a PR from @hamhands fork (but that's a total guess).

**Comment by mapsam** [comment]
@hamhands have you worked through loading in non-Mapbox basemaps at all? I'd like to see that!

I think most basemaps that require a token can have the token included in the URL so just having a single field for URL should suffice.

**Comment by hamhands** [comment]
@mapsam haven't yet but I can get going on that! Shouldn't be too big of a change.

**Comment by mapsam** [comment]
Awesome - I'll take a look at selecting the new layer as well. Keep you posted!

**Comment by mapsam** [comment]
Hey @hamhands want to switch out the `xhr` for `$.ajax()` at some point as well? Going to deprecate our own ajax function for the strong jQuery version after some talk in #191

**Comment by mapsam** [comment]
I'm getting a `Uncaught TypeError: Cannot read property '0' of undefined` when the layer adds - it doesn't seem to break anything, but any idea where it's coming from?

**Comment by hamhands** [comment]
Hmm, not sure. Maybe there's an unused variable or incorrect syntax in the execute section?

**Pull request merged**